### ISSUE TITLE
Windows MSMF not opening connected device

### ISSUE THREAD

**Issue by feliwir** [opencv/opencv#11696]
##### System information (version)
- OpenCV => 3.4.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017

##### Detailed description

Hello, i am trying to open my USB camera with MSMF (the camera is detected correctlyy by windows). The issue i am having is that vd_CaptureFormats is empty here: https://github.com/opencv/opencv/blob/3.4.1/modules/videoio/src/cap_msmf.cpp#L2311
However vd_CurrentFormats lists all supported formats.

Note that two USB cameras are connected at the same time.

##### Steps to reproduce
```c++
int main()
{
  int devices = 0;
  cv::VideoCapture cap;
  for (int i = 0; i < 5; ++i)
  {
    if (cap.open(CV_CAP_MSMF + i))
    {
      ++devices;
    }
  }
  cap.release();
}
```

**Comment by alalek** [comment]
In 3.4.1 list of formats is filtered by RGB24.

Try the latest code from 3.4 branch. MSMF backend has been refactored.

**Comment by feliwir** [comment]
@alalek
Hello thanks for the quick response. I‘d like to do that, but since we are using OpenCv in a production environment it’s not possible for me to use unstable version. Any ETA when there will be a new release with those changes?

**Comment by GregoryMorse** [comment]
Just to confirm, I am also experiencing the same issue that MSMF is not working for video capture in the latest 3.4.1 build.

**Comment by alalek** [comment]
Please try latest code from 3.4 branch. MSMF support has been significantly improved / refactored.
There is no plan to prepare separate fix for already released code from OpenCV 3.4.1.
